### ISSUE TITLE
Model inputs not working as expected

### ISSUE THREAD

**Issue by d3473r** [angular/angular#57124]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

Hi, the new model inputs are not working as expected for me.
When i change the model input from the parent component the change is not reflected in the child component.
In this example the parent component sets the `save` model input immediately to `false` again.
In the stackblitz you can the the `save` signals are getting out of sync

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-pgh9xy?file=src%2Fmain.ts

### Please provide the environment you discovered this bug in (run `ng version`)

```true
@angular/core: 18.1.1
```

### Anything else?

I want to implement a generic save component. This component sets the model input `save` to true and shows some loading indication as long as `save` is true.

This component is used from different parent component which intern either do http requests or just execute some code.
When they are finished they set the `save` model to `false` again. Then the loading animation should stop again in the child component.

**Comment by JoostK** [comment]
Although this is currently working as designed, I do agree with you on this one.

**Comment by d3473r** [comment]
With a `setTimeout` it works: https://stackblitz.com/edit/stackblitz-starters-pgsmrb?file=src%2Fmain.ts,src%2Fchild%2Fchild.component.ts

**Comment by JoostK** [comment]
Yeah, so the reason is that from a _binding_ perspective, the parent template has always evaluated to `false`, so the change detection machinery detects it as unchanged such that the `model` input is never reset. Inserting a change detection cycle where the value is flipped, before flipping it back, does cause the notification you're after.

**Comment by d3473r** [comment]
Ahh, i see, thanks for the explanation

For anyone encountering the same issue, this will work: https://stackblitz.com/edit/stackblitz-starters-nmzllx?file=src%2Fmain.ts

**Comment by nirajnandha** [comment]
https://stackblitz.com/edit/stackblitz-starters-nby8xw?file=src%2Fmain.ts,src%2Fchild%2Fchild.component.ts

Check this solution. it might help.

**Comment by richardsengers** [comment]
Not sure if this is related to this issue, but model change is also not triggered for non primitives like objects and arrays.

The model will update just fine, but the change is not triggered

For instance
```
colorRange = model.required<(string | null)[]>();

onRemoveColorRangeClick(index: number) {
    this.colorRange().splice(index, 1);
    this.colorRange.update((x) => [...x]); // need to update otherwise (colorRangeChange) event will not trigger
  }
```
  
 In the parent <component  [(colorRange)]="aValue" (colorRangeChange)="doSomething()">
 
 aValue gets updated nicely without  this.colorRange.update((x) => [...x]); but the colorRangeChange does not get triggered without

**Comment by inmativ** [comment]
> [@inmativ](https://github.com/inmativ) Please have a look at `linkedSignal` that will introduced by [#58189](https://github.com/angular/angular/pull/58189). You're using a `model` as a local state which it was not designed for.

Can it be used with banana in box?
